src/Model/Inventory/GetStockItemsData.php

```diff
@@ -0,0 +1,278 @@
+                $productIds
+            );
+        } else {
+            $select->from(
+                $this->stockIndexTableNameResolver->execute($stockId),
+                [
+                    GetStockItemsDataInterface::QUANTITY => IndexStructure::QUANTITY,
+                    GetStockItemsDataInterface::IS_SALABLE => IndexStructure::IS_SALABLE,
+                    IndexStructure::SKU
+                ]
+            )->where(
+                IndexStructure::SKU . ' IN (?)',
+                $skuArray
+            );
+        }
+
+        try {
+            foreach ($connection->fetchAll($select) as $row) {
+                if ($this->defaultStockProvider->getId() === $stockId) {
+                    $result[$productSkusById[$row['product_id']]] = [
+                        GetStockItemsDataInterface::QUANTITY => $row[GetStockItemsDataInterface::QUANTITY],
+                        GetStockItemsDataInterface::IS_SALABLE => $row[GetStockItemsDataInterface::IS_SALABLE]
+                    ];
+                } else {
+                    $result[$row[IndexStructure::SKU]] = [
+                        GetStockItemsDataInterface::QUANTITY => $row[GetStockItemsDataInterface::QUANTITY],
+                        GetStockItemsDataInterface::IS_SALABLE => $row[GetStockItemsDataInterface::IS_SALABLE]
+                    ];
+                }
+            }
+        } catch (Exception $e) {
+            throw new LocalizedException(__('Could not receive Stock Item data'), $e);
+        }
+
+        /**
+         * Fallback to the legacy cataloginventory_stock_item table.
+         * Caused by data absence in legacy cataloginventory_stock_status table
+         * for disabled products assigned to the default stock.
+         */
+        $missingStockItemSkus = [];
+
+        foreach ($skuArray as $sku) {
+            if ($result[$sku] === null) {
+                $missingStockItemSkus[] = $sku;
+            }
+        }
+
+        if (count($missingStockItemSkus)) {
+            $missingStockItemData = $this->getStockItemDataFromStockItemTable($missingStockItemSkus, $stockId);
+
+            foreach ($missingStockItemData as $sku => $row) {
+                $result[$sku] = $row;
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Retrieve stock item data for products assigned to the default stock.
+     *
+     * @param array $skuArray
+     * @param int $stockId
+     * @return array
+     */
+    protected function getStockItemDataFromStockItemTable(array $skuArray, int $stockId): array
+    {
+        $result = [];
+
+        foreach ($skuArray as $sku) {
+            $result[$sku] = null;
+        }
+
+        $skusToCheck = array_flip($skuArray);
+
+        foreach ($skuArray as $sku) {
+            if ($this->defaultStockProvider->getId() !== $stockId
+                || $this->isSingleSourceMode->execute()
+                || !$this->isSourceItemManagementAllowedForSku->execute($sku)

```

Comment:
Sometimes while fetching products in PLP we get Error and can't get the products, these 2 methods are `null` in these cases and can't execute.
In some categories everything is okay, [example](https://scandipwapmrevprtest-dev.indvp.com/default/accessoriess.html)
For other the error occurs, [example](https://scandipwapmrevprtest-dev.indvp.com/default/kids.html)

**Endpoint:** https://scandipwapmrevprtest-dev.indvp.com/graphql

**Full  GraphQL Query:**
`query (
	$sort_1: ProductAttributeSortInput!
	$filter_1: ProductAttributeFilterInput!
	$pageSize_1: Int!
	$currentPage_1: Int!
) {
	products(
		sort: $sort_1
		filter: $filter_1
		pageSize: $pageSize_1
		currentPage: $currentPage_1
	) {
		total_count
		items {
			uid
			id
			sku
			name
			type_id
			stock_status
			salable_qty
			stock_item {
				min_sale_qty
				max_sale_qty
				qty_increments
			}
			price_range {
				minimum_price {
					discount {
						amount_off
						percent_off
					}
					final_price {
						currency
						value
					}
					final_price_excl_tax {
						currency
						value
					}
					regular_price {
						currency
						value
					}
					regular_price_excl_tax {
						currency
						value
					}
					default_price {
						currency
						value
					}
					default_final_price {
						currency
						value
					}
					default_final_price_excl_tax {
						currency
						value
					}
				}
				maximum_price {
					discount {
						amount_off
						percent_off
					}
					final_price {
						currency
						value
					}
					final_price_excl_tax {
						currency
						value
					}
					regular_price {
						currency
						value
					}
					regular_price_excl_tax {
						currency
						value
					}
					default_price {
						currency
						value
					}
					default_final_price {
						currency
						value
					}
					default_final_price_excl_tax {
						currency
						value
					}
				}
			}
			image {
				path
				url
			}
			thumbnail {
				path
				url
			}
			small_image {
				path
				url
			}
			short_description {
				html
			}
			special_from_date
			special_to_date
			price_tiers {
				discount {
					amount_off
					percent_off
				}
				final_price {
					currency
					value
				}
				quantity
			}
			attributes: s_attributes {
				attribute_id
				attribute_value
				attribute_code
				attribute_label
				attribute_type
				attribute_group_id
				attribute_group_name
				attribute_options {
					label
					value
					swatch_data {
						type
						value
					}
				}
			}
			url
			url_rewrites {
				url
			}
			review_count
			rating_summary
			... on CustomizableProductInterface {
				options {
					... on CustomizableDropDownOption {
						dropdownValues: value {
							uid
							option_type_id
							price
							priceInclTax
							priceExclTax
							price_type
							currency
							sku
							title
							sort_order
						}
					}
					... on CustomizableRadioOption {
						dropdownValues: value {
							uid
							option_type_id
							price
							priceInclTax
							priceExclTax
							price_type
							currency
							sku
							title
							sort_order
						}
					}
					... on CustomizableCheckboxOption {
						checkboxValues: value {
							uid
							option_type_id
							price
							priceInclTax
							priceExclTax
							price_type
							currency
							sku
							title
							sort_order
						}
					}
					... on CustomizableMultipleOption {
						checkboxValues: value {
							uid
							option_type_id
							price
							priceInclTax
							priceExclTax
							price_type
							currency
							sku
							title
							sort_order
						}
					}
					... on CustomizableFieldOption {
						fieldValues: value {
							price
							priceInclTax
							priceExclTax
							price_type
							currency
							sku
							max_characters
						}
						product_sku
					}
					... on CustomizableAreaOption {
						areaValues: value {
							price
							priceInclTax
							priceExclTax
							price_type
							currency
							sku
							max_characters
						}
						product_sku
					}
					... on CustomizableFileOption {
						fileValues: value {
							price
							priceInclTax
							priceExclTax
							price_type
							currency
							sku
							file_extension
						}
					}
					... on CustomizableDateOption {
						value {
							price
							priceInclTax
							priceExclTax
							price_type
							currency
							sku
						}
						product_sku
					}
					title
					required
					sort_order
					type
					uid
				}
			}
			... on ConfigurableProduct {
				configurable_options {
					attribute_code
					values {
						value_index
					}
				}
				variants: variants_plp {
					product {
						uid
						id
						sku
						name
						type_id
						stock_status
						salable_qty
						stock_item {
							min_sale_qty
							max_sale_qty
							qty_increments
						}
						price_range {
							minimum_price {
								discount {
									amount_off
									percent_off
								}
								final_price {
									currency
									value
								}
								final_price_excl_tax {
									currency
									value
								}
								regular_price {
									currency
									value
								}
								regular_price_excl_tax {
									currency
									value
								}
								default_price {
									currency
									value
								}
								default_final_price {
									currency
									value
								}
								default_final_price_excl_tax {
									currency
									value
								}
							}
							maximum_price {
								discount {
									amount_off
									percent_off
								}
								final_price {
									currency
									value
								}
								final_price_excl_tax {
									currency
									value
								}
								regular_price {
									currency
									value
								}
								regular_price_excl_tax {
									currency
									value
								}
								default_price {
									currency
									value
								}
								default_final_price {
									currency
									value
								}
								default_final_price_excl_tax {
									currency
									value
								}
							}
						}
						attributes: s_attributes {
							attribute_id
							attribute_value
							attribute_code
							attribute_label
							attribute_type
							attribute_group_id
							attribute_group_name
						}
					}
				}
			}
			... on BundleProduct {
				dynamic_price
				dynamic_sku
				ship_bundle_items
				dynamic_weight
				items {
					uid
					option_id
					title
					required
					type
					position
					sku
					options {
						uid
						label
						quantity
						position
						is_default
						price
						price_type
						can_change_quantity
						product {
							name
							stock_status
							price_range {
								minimum_price {
									discount {
										amount_off
										percent_off
									}
									final_price {
										currency
										value
									}
									final_price_excl_tax {
										currency
										value
									}
									regular_price {
										currency
										value
									}
									regular_price_excl_tax {
										currency
										value
									}
									default_price {
										currency
										value
									}
									default_final_price {
										currency
										value
									}
									default_final_price_excl_tax {
										currency
										value
									}
								}
								maximum_price {
									discount {
										amount_off
										percent_off
									}
									final_price {
										currency
										value
									}
									final_price_excl_tax {
										currency
										value
									}
									regular_price {
										currency
										value
									}
									regular_price_excl_tax {
										currency
										value
									}
									default_price {
										currency
										value
									}
									default_final_price {
										currency
										value
									}
									default_final_price_excl_tax {
										currency
										value
									}
								}
							}
						}
					}
				}
				bundle_options {
					option_id
					selection_details {
						selection_id
						final_option_price
						final_option_price_excl_tax
						regular_option_price
						regular_option_price_excl_tax
					}
				}
			}
			... on GroupedProduct {
				items {
					product {
						uid
						id
						sku
						name
						type_id
						stock_status
						salable_qty
						stock_item {
							min_sale_qty
							max_sale_qty
							qty_increments
						}
						price_range {
							minimum_price {
								discount {
									amount_off
									percent_off
								}
								final_price {
									currency
									value
								}
								final_price_excl_tax {
									currency
									value
								}
								regular_price {
									currency
									value
								}
								regular_price_excl_tax {
									currency
									value
								}
								default_price {
									currency
									value
								}
								default_final_price {
									currency
									value
								}
								default_final_price_excl_tax {
									currency
									value
								}
							}
							maximum_price {
								discount {
									amount_off
									percent_off
								}
								final_price {
									currency
									value
								}
								final_price_excl_tax {
									currency
									value
								}
								regular_price {
									currency
									value
								}
								regular_price_excl_tax {
									currency
									value
								}
								default_price {
									currency
									value
								}
								default_final_price {
									currency
									value
								}
								default_final_price_excl_tax {
									currency
									value
								}
							}
						}
						attributes: s_attributes {
							attribute_id
							attribute_value
							attribute_code
							attribute_label
							attribute_type
							attribute_group_id
							attribute_group_name
						}
					}
					position
					qty
				}
			}
			... on DownloadableProduct {
				links_purchased_separately
			}
		}
		page_info {
			current_page
			total_pages
		}
	}
}
`

**Variables:**
`{
	"sort_1": {
		"name": "ASC"
	},
	"filter_1": {
		"price": {},
		"category_id": {
			"eq": 40
		},
		"customer_group_id": {
			"eq": "0"
		}
	},
	"pageSize_1": 24,
	"currentPage_1": 1,
	"_currency": ""
}`